### PR TITLE
Yrax Quadraphracts

### PR DESCRIPTION
**Pull request by John-Candlebury**
#### Summary
Content "Add a new high danger Yrax construct"

#### Purpose of change

I wanted some very low intensity weirdness to be possible after the player completes and meets the spoiler creature at the end of physics microlabs, and this here is the low intensity weirdness.
 
Spider tanks are cool and I wanted to add one ever since I noticed they were possible.

#### Describe the solution

Quadraphracts are "pretend multi tile" monsters, they consist of a single main body monster that does most things other than melee combat and four leg monsters that are teleported together with the main body. Both types of monsters share HP relationships, so its possible to kill the whole thing by disabling legs, and the legs die if you deal lethal damage to the main body (not that this is easy). This all works through actually not very complex EOCs.

The Quadraphracts have three attacks that can be chosen depending on the amount of nearby foes, they either:

- Use their main cannon to destroy enemies at large distance like tankbots used too.
- Deploy defensive manhack equivalents at close range.
- Overload and unlocks new attacks from these man-hacks if they are getting overrun.
- Deploy smoke mortars at mid range, they can still see the player through smokes. (I wanted to make them all_seeing but this resulted in them getting stuck fighting enemies in basements).

Like all Yrax they are initially neutral to the player. 

#### Describe alternatives you've considered
Add hostile ones somewhere the Yrax don't want you near in the future.
#### Testing

"Fought" a few of these but mostly saw them stomp on zombies.

#### Additional context

This isn't even the Yrax equivalent to an MBT

### CODE REVIEW

File: data/json/monsters/yrax.json [hunk 1]
```diff
@@ -215,6 +215,94 @@
...
+    "hp": 400,
+    "proportional": { "dodge": 0.25, "speed": 1.2 },
+    "melee_dice": 6,
+    "melee_dice_sides": 15,
+    "melee_skill": 4,
+    "special_attacks": [ { "id": "smash", "throw_strength": 96 } ],
+    "armor": { "bash": 1000, "cut": 2000, "acid": 100, "heat": 500, "electric": 100, "bullet": 1000 },
+    "death_function": {
+      "corpse_type": "BROKEN",
+      "message": "The quadraphract stumbles as one of its legs looses coherence and collapsed.",
```
Comment (harakka) [inline data/json/monsters/yrax.json:299]:
```suggestion
      "message": "The quadraphract stumbles as one of its legs looses coherence and collapses.",
```

**Comment by KittyTac** [comment]
I hope this paves the way for e.g multi-tile "normal" giant spiders!

**Comment by I-am-Erk** [comment]
I definitely need to work that giant monster code into some kind of boss for this dungeon

**Comment by John-Candlebury** [comment]
Can help you with a HG wells style combat tripod. That would be cool.

**Comment by I-am-Erk** [comment]
Oh that would be *perfect*

**Pull request merged**